shhudspeth/BookRecommendationSystem
# Split Training and Test data in Utility Matrix

What is the purpose of this PR and why do we need it?
I updated the KNN Utility Matrix Recommender system with a train-test split and found it's RMSE metric. 

How did you solve the problem? What are the benefits and drawbacks?
I manually (versus used sklearn or other python library) the data and loaded it into the KNN model. Since it we pivoted and changed original dataFrame into a sparse matrix there was some finagling of data shapes that had to happen in ordered to generate the predicted values. 

I then also create my own function to determine RMSE. I did not get a chance to put this all into more deployment code as i felt I was still in an analysis stage, but I have a strong idea of what  I need to do. 

Because of the "funny" data shapes, I reduced the amount of data I used to find the RMSE-- if I had used all of the testing data, I'm sure I'd have a strong RMSE, but I haven't optimized my self-created code to make that computation efficient. 

How do we test it?
I tested as I built it. Running the scripts should show some of the checks I used. 

Are there any potential side affects?
If we take a larger portion of the test data to build the RSME, it will take a long time to run.
